authors/apps/articles/views.py

```diff
@@ -0,0 +1,104 @@
+import json
+
+from django.conf import settings
+from django.contrib.sites.shortcuts import get_current_site
+from django.core.mail import send_mail
+from django.shortcuts import get_object_or_404
+from django.urls import reverse
+from django.utils.encoding import force_bytes
+from django.utils.encoding import force_text
+from django.utils.http import urlsafe_base64_encode, urlsafe_base64_decode
+from rest_framework import status
+from rest_framework.generics import GenericAPIView, CreateAPIView, UpdateAPIView
+from rest_framework.generics import RetrieveUpdateAPIView, RetrieveUpdateDestroyAPIView
+from rest_framework.permissions import AllowAny, IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.exceptions import NotFound
+
+from config.settings import default
+from .models import Article
+from .renderers import ArticleJSONRenderer
+from .serializers import (
+     ArticleSerializer
+)
+
+from authors.apps.profiles.models import Profile
+from authors.apps.authentication.models import User
+
+
+class ArticleView(APIView):
+    """creating, viewing , deleting and updating articles"""
+
+    permission_classes = (IsAuthenticated,)
+
+
+    def post(self, request):
+        serializer_context = {'author': request.user.profile}
+        article = request.data.get('article', {})
+
+        # Create an article from the above data
+        serializer = ArticleSerializer(data=article, context=serializer_context)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        return Response({"articles": serializer.data}, status=status.HTTP_201_CREATED)
```

Comment:
@frankopkusianwar , instead of creating a response json, add a `renderer class` to the view